### PR TITLE
Overpressure - Add editor range attribute for vehicles

### PR DESCRIPTION
**Pull request by bluefieldcreator**
**When merged this pull request will:**
- Add overpressure range setting per vehicle

~~**Problem**: if the user has the server setting set to 10 (example) and a mission maker wants to set a vehicle's value to 0, it wont work. Since we check for >= 0.1~~

~~I cant figure out a good system other than making an entire new overpressure setting called "enable overpressure" as a checkbox instead of the range slider.~~

~~There might be _some_ use cases where a user may want to disable overpressure on some vehicles, and change the values of others, which would require both... so I might update this in the future to add such use case.~~

**Important bits of code: cfgEden.hpp & fnc_firedEHOP.sqf**

2nd ever PR i appreciate all feedback. 🥲 

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/overpressure/functions/fnc_firedEHOP.sqf [hunk 1]
```diff
@@ -23,10 +23,9 @@ private _opValues = [_weapon, _ammo, _magazine] call FUNC(getOverPressureValues)
 
 _opValues params ["_dangerZoneAngle", "_dangerZoneRange", "_dangerZoneDamage"];
 
-private _unitCoef = _unit getVariable [QEGVAR(overpressure,distance), GVAR(overpressureDistanceCoefficient)];
-systemChat format ["coef  = %1", _unitCoef];
+private _unitCoef = _unit getVariable [QEGVAR(overpressure,distance), GVAR(overpressureDistanceCoefficient)]; // Because server setting is <null> we fall back to server settings.
```
Comment (bluefieldcreator) [inline addons/overpressure/functions/fnc_firedEHOP.sqf:26]: Meant to say "becasue attribute setting is <null> ..."

**Comment by bluefieldcreator** [comment]
Why are all the tabs failing OH MY GOD

**Comment by bluefieldcreator** [comment]
TODO:
- FIX TEXT IN ATTRIBUTES
- CHANGE SLIDER TO TEXT BOX
- ADD DEFAULT -1 VALUE TO CHECK AGAINST ENABLING, DISABLING AND CUSTOM VALUES

File: addons/overpressure/cfgEden.hpp [hunk 2]
```diff
@@ -1,46 +1,18 @@
...
-        };
-    };
     class Object {
         class AttributeCategories {
             class ace_attributes {
                 class Attributes {
                     class GVAR(overPressureDistanceSetting) {
-                        property = QUOTE(overpressureDistanceCoefficient);
-                        control = QGVAR(slider);
                         displayName = CSTRING(Overpressure_distanceCoefficient_displayName);
```
Comment (bluefieldcreator) [inline addons/overpressure/cfgEden.hpp:7]: I dont know why this is not working. CSTRING should check the stringtables, right?

Comment (bluefieldcreator) [reply addons/overpressure/cfgEden.hpp:7]: Rn its empty in the attributes list, kinda annoying.

Comment (bluefieldcreator) [reply addons/overpressure/cfgEden.hpp:7]: image

**Comment by bluefieldcreator** [comment]
Okay, I fixed EVERYTHING - This PR is ready for review & merge.

**Comment by bluefieldcreator** [comment]
Good catch

File: addons/overpressure/functions/fnc_firedEHOP.sqf [hunk 3]
```diff
@@ -22,7 +22,11 @@ TRACE_8("firedEH:",_unit,_weapon,_muzzle,_mode,_ammo,_magazine,_projectile,_gunn
 private _opValues = [_weapon, _ammo, _magazine] call FUNC(getOverPressureValues);
 
 _opValues params ["_dangerZoneAngle", "_dangerZoneRange", "_dangerZoneDamage"];
-_dangerZoneRange = _dangerZoneRange * GVAR(overpressureDistanceCoefficient);
+
+private _unitOverpressureRangeAttribute = _unit getVariable QEGVAR(overpressure,distance);
+
+// If -1 (disabled) we return to server value, otherwise we use the given value.
+_dangerZoneRange = _dangerZoneRange * [_unitOverpressureRangeAttribute, GVAR(overpressureDistanceCoefficient)] select (_unitOverpressureRangeAttribute < 0);
```
Comment (PabstMirror) [inline addons/overpressure/functions/fnc_firedEHOP.sqf:29]: help[L-S12]: Invalid Args - [B:*]
   ┌─ addons/overpressure/functions/fnc_firedEHOP.sqf:29:37

you are multiplying a number times an array

File: addons/overpressure/functions/fnc_firedEHOP.sqf [hunk 4]
```diff
@@ -22,7 +22,11 @@ TRACE_8("firedEH:",_unit,_weapon,_muzzle,_mode,_ammo,_magazine,_projectile,_gunn
 private _opValues = [_weapon, _ammo, _magazine] call FUNC(getOverPressureValues);
 
 _opValues params ["_dangerZoneAngle", "_dangerZoneRange", "_dangerZoneDamage"];
-_dangerZoneRange = _dangerZoneRange * GVAR(overpressureDistanceCoefficient);
+
+private _unitOverpressureRangeAttribute = _unit getVariable QEGVAR(overpressure,distance);
```
Comment (PabstMirror) [inline addons/overpressure/functions/fnc_firedEHOP.sqf:26]: this will be undefined for most cases

**Review by PabstMirror — CHANGES_REQUESTED** [review]
not working

File: addons/overpressure/CfgEden.hpp [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    class Object {
+        class AttributeCategories {
+            class ace_attributes {
+                class Attributes {
+                    class GVAR(overPressureDistanceSetting) {
+                        displayName = CSTRING(distanceCoefficient_displayName);
+                        tooltip = CSTRING(distanceCoefficient_toolTip);
+                        property = QUOTE(overpressureDistanceCoefficient);
+                        control = "Edit";
+                        expression = QUOTE(_this setVariable [ARR_3(QQEGVAR(overpressure,distance),_value,true)]);
```
Comment (PabstMirror) [inline addons/overpressure/CfgEden.hpp:11]: small optimization to not sync var when it's default
```suggestion
                        expression = QUOTE(if (_value >= 0) then { _this setVariable [ARR_3(QQGVAR(distance),_value,true)] });
```

File: addons/overpressure/functions/fnc_firedEHOP.sqf [hunk 6]
```diff
@@ -22,7 +22,11 @@ TRACE_8("firedEH:",_unit,_weapon,_muzzle,_mode,_ammo,_magazine,_projectile,_gunn
 private _opValues = [_weapon, _ammo, _magazine] call FUNC(getOverPressureValues);
 
 _opValues params ["_dangerZoneAngle", "_dangerZoneRange", "_dangerZoneDamage"];
-_dangerZoneRange = _dangerZoneRange * GVAR(overpressureDistanceCoefficient);
+
+private _unitOverpressureRangeAttribute = _unit getVariable [QEGVAR(overpressure,distance), -1];
```
Comment (PabstMirror) [inline addons/overpressure/functions/fnc_firedEHOP.sqf:26]:
```suggestion
private _unitOverpressureRangeAttribute = _unit getVariable [QEGVAR(overpressure,distance), GVAR(overpressureDistanceCoefficient)];
```

File: addons/overpressure/functions/fnc_firedEHOP.sqf [hunk 7]
```diff
@@ -22,7 +22,11 @@ TRACE_8("firedEH:",_unit,_weapon,_muzzle,_mode,_ammo,_magazine,_projectile,_gunn
...
+
+// If -1 (disabled) we return to server value, otherwise we use the given value.
+_dangerZoneRange = _dangerZoneRange * ([_unitOverpressureRangeAttribute, GVAR(overpressureDistanceCoefficient)] select (_unitOverpressureRangeAttribute < 0));
```
Comment (PabstMirror) [inline addons/overpressure/functions/fnc_firedEHOP.sqf:29]:
```suggestion
_dangerZoneRange = _dangerZoneRange * _unitOverpressureRangeAttribute;
```

File: addons/overpressure/CfgEden.hpp (hunk 5 above)
Comment (johnb432) [reply addons/overpressure/CfgEden.hpp:11]: @bluefieldcreator Why were these marked as resolved, when they haven't been?

File: addons/overpressure/config.cpp [hunk 8]
```diff
@@ -8,7 +8,7 @@ class CfgPatches {
         requiredVersion = REQUIRED_VERSION;
         requiredAddons[] = {"ace_common"};
         author = ECSTRING(common,ACETeam);
-        authors[] = {"commy2","KoffeinFlummi","esteldunedain"};
+        authors[] = {"commy2","KoffeinFlummi","esteldunedain", "bluefield"};
```
Comment (johnb432) [inline addons/overpressure/config.cpp:11]: IMO this is reserved for people who have contributed significantly in either the creation or the refactoring of a component, https://github.com/acemod/ACE3/blob/master/AUTHORS.txt would be more appropriate for this PR.

File: addons/overpressure/CfgEden.hpp (hunk 5 above)
Comment (bluefieldcreator) [reply addons/overpressure/CfgEden.hpp:11]: @johnb432 My bad, the GitHub phone app is ass

**Comment by bluefieldcreator** [comment]
I think... that's it... I hate the GH phone app, I thought it merged your changes.

My bad

File: addons/overpressure/functions/fnc_firedEHOP.sqf (hunk 7 above)
Comment (johnb432) [reply addons/overpressure/functions/fnc_firedEHOP.sqf]: @bluefieldcreator Can you merge this too?

File: addons/overpressure/functions/fnc_firedEHOP.sqf [hunk 9]
```diff
@@ -22,7 +22,10 @@ TRACE_8("firedEH:",_unit,_weapon,_muzzle,_mode,_ammo,_magazine,_projectile,_gunn
 private _opValues = [_weapon, _ammo, _magazine] call FUNC(getOverPressureValues);
 
 _opValues params ["_dangerZoneAngle", "_dangerZoneRange", "_dangerZoneDamage"];
-_dangerZoneRange = _dangerZoneRange * GVAR(overpressureDistanceCoefficient);
+
+private _unitOverpressureRangeAttribute = _unit getVariable [QEGVAR(overpressure,distance), GVAR(overpressureDistanceCoefficient)];
+
+_dangerZoneRange = _dangerZoneRange * ([_unitOverpressureRangeAttribute, GVAR(overpressureDistanceCoefficient)] select (_unitOverpressureRangeAttribute < 0));
 
 TRACE_3("cache",_dangerZoneAngle,_dangerZoneRange,_dangerZoneDamage);
```
Comment (PabstMirror) [inline addons/overpressure/functions/fnc_firedEHOP.sqf:30]:
```
TRACE_3("cache",_dangerZoneAngle,_dangerZoneRange,_dangerZoneDamage);

if (_dangerZoneDamage <= 0) exitWith {};

private _unitOverpressureRangeAttribute = _unit getVariable [QGVAR(distance), GVAR(overpressureDistanceCoefficient)];
_dangerZoneRange = _dangerZoneRange * _unitOverpressureRangeAttribute;

// The weapon produces overpressure, calculate
```
moving the distance calcs bellow the damage exitWith is a nice small optimization
I can't do this in a suggestion but we can always just merge as-is and fix in a followup

**Review by PabstMirror — APPROVED** [review]

**Review by PabstMirror — APPROVED** [review]

**Comment by bluefieldcreator** [comment]
Finally... :D

**Pull request merged**